tiledb/sm/filter/filter_buffer.h

```diff
@@ -339,7 +339,7 @@ class FilterBuffer {
     shared_ptr<Buffer> underlying_buffer_;
 
     /** True if this instance is a view on the underlying buffer. */
-    bool is_view_;
+    bool is_view_{false};
```

Comment:
Instead of adding this value, should we disable the default constructor? I believe the default constructor is not valid for the `BufferOrView` class?

Reply:
Does this class actually have default constructor (I am not 100% sure, but we defined custom constructors so the default one should be implicitly deleted)? I put this value here so that in the move constructor the `is_view_` value will be initialized.

Reply:
> we defined custom constructors so the default one should be implicitly deleted

Indeed, default constructors [are not](https://en.cppreference.com/w/cpp/language/default_constructor#:~:text=If%20there%20is%20no%20user%2Ddeclared%20constructor%20or%20constructor%20template%20for%20a%20class%20type%2C%20the%20compiler%20will%20implicitly%20declare%20a%20default%20constructor) implicitly declared when there other constructors are manually declared (TIL).